### ISSUE TITLE
rerouteToError causes NPE in handleTriggeredBeforeEvent

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#17597]
### Description of the bug

A BeforeEventListener which attempts to reroute to the error screen will cause Flow to fail with a NPE in AbstractNavigationStateRenderer.

This worked fine in Vaadin 24.1.8 but started to fail in Vaadin 24.1.9.

Stacktrace:
```
java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "com.vaadin.flow.router.NavigationState.getResolvedPath()" is null
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.handleTriggeredBeforeEvent(AbstractNavigationStateRenderer.java:672)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.sendBeforeEnterEvent(AbstractNavigationStateRenderer.java:602)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.sendBeforeEnterEvent(AbstractNavigationStateRenderer.java:577)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.createChainIfEmptyAndExecuteBeforeEnterNavigation(AbstractNavigationStateRenderer.java:453)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.handle(AbstractNavigationStateRenderer.java:210)
	at com.vaadin.flow.router.internal.ErrorStateRenderer.handle(ErrorStateRenderer.java:106)
	at com.vaadin.flow.component.UI.handleExceptionNavigation(UI.java:1890)
	at com.vaadin.flow.component.UI.handleNavigation(UI.java:1865)
	at com.vaadin.flow.component.UI.navigate(UI.java:1246)
	at com.vaadin.flow.component.UI.navigate(UI.java:1201)
	at com.vaadin.flow.component.UI.navigate(UI.java:1004)
	at com.vaadin.flow.component.UI.navigate(UI.java:897)
	at com.vaadin.starter.skeleton.MainView.lambda$new$9b1b5227$1(MainView.java:23)
	at com.vaadin.flow.component.ComponentEventBus.fireEventForListener(ComponentEventBus.java:233)
```

### Expected behavior

This should succeed and `InternalServerError` should be rendered.

### Minimal reproducible example

Run the Skeleton Starter with the following `MainView`:
```java
@Route("")
public class MainView extends VerticalLayout {

    public MainView() {
        UI.getCurrent().addBeforeEnterListener(e -> {
           e.rerouteToError(new RuntimeException("Simulated"), "Simulated");
        });
        add(new Button("Simulate", e -> UI.getCurrent().navigate(MainView.class)));
    }
}
```
Press the "Simulate" button - the exception is now logged in stdout.

### Versions

- Vaadin / Flow version: 24.1.9, flow 24.1.11
- Java version: 17
- OS version: Ubuntu

**Comment by mcollovati** [comment]
Should have been fixed by #17587

**Comment by mvysny** [comment]
Awesome, thank you for such a quick fix 🙇 🥳

**Comment by mvysny** [comment]
Looks to be fixed in Vaadin 24.1.10, thank you :partying_face:
